cmd/jaeger/internal/extension/jaegerquery/server_test.go

```diff
@@ -0,0 +1,308 @@
+		},
+		{
+			name: "dependency error",
+			config: &Config{
+				TraceStorageArchive: "jaeger_storage",
+				TraceStoragePrimary: "need-dependency-reader-error",
+			},
+			expectedErr: "cannot create dependencies reader",
+		},
+		{
+			name: "storage archive error",
+			config: &Config{
+				TraceStorageArchive: "need-factory-error",
+				TraceStoragePrimary: "jaeger_storage",
+			},
+			expectedErr: "cannot find archive storage factory",
+		},
+		{
+			name: "tenacy manager error",
+			config: &Config{
+				TraceStorageArchive: "jaeger_storage",
+				TraceStoragePrimary: "jaeger_storage",
+				Tenancy: tenancy.Options{
+					Enabled: true,
+					Header:  "test",
+				},
+			},
+			expectedErr: "could not start jaeger-query",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			telemetrySettings := component.TelemetrySettings{
+				Logger: logger,
+			}
+			server := newServer(tt.config, telemetrySettings)
+			err := server.Start(context.Background(), host)
+
+			if tt.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.ErrorContains(t, err, tt.expectedErr)
+			}
+		})
+	}
+}
+
+func TestServerAddArchiveStorage(t *testing.T) {
+	host := componenttest.NewNopHost()
+
+	tests := []struct {
+		name           string
+		qSvcOpts       *querysvc.QueryServiceOptions
+		config         *Config
+		extension      component.Component
+		expectedOutput string
+		expectedErr    string
+	}{
+		{
+			name:           "Archive storage unset",
+			config:         &Config{},
+			qSvcOpts:       &querysvc.QueryServiceOptions{},
+			expectedOutput: `{"level":"info","msg":"Archive storage not configured"}` + "\n",
+			expectedErr:    "",
+		},
+		{
+			name: "Archive storage set",
+			config: &Config{
+				TraceStorageArchive: "random-value",
+			},
+			qSvcOpts:       &querysvc.QueryServiceOptions{},
+			expectedOutput: "",
+			expectedErr:    "cannot find archive storage factory: cannot find extension",
+		},
+		{
+			name: "Archive storage not supported",
+			config: &Config{
+				TraceStorageArchive: "badger",
+			},
+			qSvcOpts:       &querysvc.QueryServiceOptions{},
+			extension:      fakeStorageExt{},
+			expectedOutput: "Archive storage not supported by the factory",
+			expectedErr:    "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			logger, buf := testutils.NewLogger()
+			telemetrySettings := component.TelemetrySettings{
+				Logger: logger,
+			}
+			server := newServer(tt.config, telemetrySettings)
+			if tt.extension != nil {
+				host = storageHost{extension: tt.extension}
+			}
+			err := server.addArchiveStorage(tt.qSvcOpts, host)
+			if tt.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.ErrorContains(t, err, tt.expectedErr)
+			}
+
+			assert.Contains(t, buf.String(), tt.expectedOutput)
+		})
+	}
+}
+
+func TestServerShutdown(t *testing.T) {
+	tests := []struct {
+		name           string
+		hasQueryServer bool
+	}{
```

Comment:
there is no benefit in making this a data driven test. This would be more readable:
```
t.Run("without query server", (t *testing.T) { /* first test /* })
t.Run("without query server", (t *testing.T) { /* 2nd test /* })
```
You can reuse telemetry settings across these, but aside from that there's little overlap